### PR TITLE
Style refactoring logo styles

### PR DESCRIPTION
**Pull request by marharyta**
# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [ ] Link to issue - no particular issue, we agreed with @Macroz I will do some styles refactoring next
- [x] Note if PR is on top of other PR - [https://github.com/CSCfi/rems/pull/2483](https://github.com/CSCfi/rems/pull/2483)
- [ ] Note if related change in rems-deploy repo
- [ ] Consider adding screenshots for ease of review - should not have any visible effect, just refactoring

### CODE REVIEW

File: src/cljs/rems/spa.cljs
Comment (Macroz) [inline src/cljs/rems/spa.cljs:101]: We should rather use the existing sub `:language` (from `user_settings.cljs`). This will not work.

File: src/cljs/rems/navbar.cljs
Comment (Macroz) [inline src/cljs/rems/navbar.cljs:65]: This is not the best place for the `logo` component (`navbar.cljs`), `atoms.cljs` would perhaps be better.

File: src/cljs/rems/spa.cljs
Comment (Macroz) [inline src/cljs/rems/spa.cljs:347]: This logic is still wrong. The regular logo should be there always when configured so. It should not depend on the page id or navbar logo.

**Review by Macroz — CHANGES_REQUESTED** [review]
Looks pretty good but there are a two issues still that need resolving before this can be merged.

File: src/cljs/rems/navbar.cljs
Comment (marharyta) [reply src/cljs/rems/navbar.cljs:65]: agree

**Comment by opqdonut** [comment]
did you mean to review #2483 instead @Macroz? of course if this PR is fine too it's easier to merge this one instead of first #2483 and then this...

**Comment by Macroz** [comment]
> did you mean to review #2483 instead @Macroz? of course if this PR is fine too it's easier to merge this one instead of first #2483 and then this...

Yeah I didn't understand that this is a new PR on top of that old. These issues need to be resolved in the original PR and then this can be rebased and reviewed again. I don't really see the point of doing this second PR with the open issues remaining.

**Comment by marharyta** [comment]
So, @opqdonut @Macroz is it good to go?

File: src/clj/rems/css/styles.clj
Comment (opqdonut) [inline src/clj/rems/css/styles.clj:275]: were you thinking of consistency with the `generate-foo` functions when you turned these from constants to functions? If so, should we use the `generate-` prefix for everything, or perhaps for nothing?

File: src/clj/rems/css/styles.clj
Comment (opqdonut) [inline src/clj/rems/css/styles.clj:127]: Do we need separate `generate-logo-media-queries` and `generate-logo-styles`? Could they just be one chunk since they're all related to the logos? Or is there some reason why we need to have `generate-logo-styles` early in the main stylesheet and `generate-logo-media-queries` late?

**Review by opqdonut — COMMENTED** [review]
some questions

File: src/clj/rems/css/styles.clj
Comment (opqdonut) [reply src/clj/rems/css/styles.clj:275]: I think it makes sense to use functions for these style fragments, because they might need to look up stuff from the config/theme. Mixing constants and functions can lead to surprises when developing, so maybe let's just make everything a function. At that point, I think the `generate-` prefixes can be dropped from the names, it's just noise.

Comment (marharyta) [reply src/clj/rems/css/styles.clj:275]: I just remember we did have a discussion with @Macroz the other day we decided to get rid of the ```private``` definitions

File: src/clj/rems/css/styles.clj
Comment (marharyta) [reply src/clj/rems/css/styles.clj:127]: It is considered to be a good practice to separate media rules into separate mixins, in our case: functions

File: src/clj/rems/css/styles.clj
Comment (marharyta) [reply src/clj/rems/css/styles.clj:275]: I think this is a good point naming convention-wise. I can fix that @opqdonut

Comment (marharyta) [reply src/clj/rems/css/styles.clj]: Well, if we want to to drop it all together then we have to either separate then into mixins file and definitely do it in a standalone PR

Comment (Macroz) [reply src/clj/rems/css/styles.clj]: I think I remember the discussion. I was actually not suggesting then to replace these `def` with `defn`. There is no need to mark them as private here really. But also I was thinking about a case where one does `defn` and `^:private` which does not need to be so because `defn-` exists.

File: src/clj/rems/css/styles.clj
Comment (Macroz) [reply src/clj/rems/css/styles.clj:127]: I think these are two separate things. The articles you linked suggest putting the media queries into the component i.e. logo here. They suggest writing those media queries with reusable helpers and not by hand. What you are doing here is not creating a reusable media query mixing, you are simply splitting the code into two places.

Comment (Macroz) [reply src/clj/rems/css/styles.clj:127]: I suggest that you put the logo related stuff including styles and media queries into one function `generate-logo-styles` and remove the `generate-logo-media-queries. We can later refactor the media queries themselves.

**Review by opqdonut — APPROVED** [review]

**Comment by marharyta** [comment]
@Macroz Cannot merge it, can you check wether it is now according to the spec?

**Comment by marharyta** [comment]
I cannot merge since you have blocked it with the request changes

**Pull request merged**
